### PR TITLE
Standalone node on PowerVS

### PR DESCRIPTION
**Pull request by sudeeshjohn**
This is a modified code of PR  https://github.com/ocp-power-automation/infra/pull/7

### CODE REVIEW

File: infra-node/LICENCE.txt [hunk 1]
```diff
@@ -0,0 +1,201 @@
+                                 Apache License
```
Comment (mkumatag) [inline infra-node/LICENCE.txt:1]: not needed as parent dir already got a license file

File: infra-node/data/.gitignore [hunk 2]
```diff
@@ -0,0 +1,4 @@
+# Ignore everything in this directory
```
Comment (mkumatag) [inline infra-node/data/.gitignore:1]: this can be removed

File: infra-node/infra-node.tf [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+resource "random_id" "label" {
+    count = var.vm_id == "" ? 1 : 0
+    byte_length = "2" # Since we use the hex, the word lenght would double
+    prefix = "${var.vm_id_prefix}-"
+}
+
+locals {
+    # Generates vm_id as combination of vm_id_prefix + (random_id or user-defined vm_id)
+    vm_id      = var.vm_id == "" ? random_id.label[0].hex : "${var.vm_id_prefix}-${var.vm_id}"
```
Comment (yussufsh) [inline infra-node/infra-node.tf:15]: Prefix is not getting added if vm_id is "".

File: infra-node/infra-node.tf [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    public_key                      = local.public_key
+    ssh_agent                       = var.ssh_agent
+    rhel_subscription_username      = var.rhel_subscription_username
+    rhel_subscription_password      = var.rhel_subscription_password
+    rhel_smt                        = var.rhel_smt
+    storage_type                    = ""
+    volume_type                     = ""
+    volume_size                     = ""
+    volume_shareable                = ""
+    setup_squid_proxy               = true
```
Comment (yussufsh) [inline infra-node/infra-node.tf:42]: false here?

File: infra-node/outputs.tf [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    value = join(", ", module.prepare.bastion_ip)
+}
+
+output "bastion_public_ip" {
+    value = join(", ", module.prepare.bastion_public_ip)
+}
+
+output "bastion_ssh_command" {
+    value = join(", ", formatlist("ssh -i ${var.private_key_file} ${var.rhel_username}@%s", module.prepare.bastion_public_ip))
+}
```
Comment (yussufsh) [inline infra-node/outputs.tf:35]: newline

File: infra-node/var.tfvars [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+### Example 0.5 processors == 0.5 physical core entitlements == ceil[0.5] = 1 vCPU == 8 logical OS CPUs (SMT=8)
+### Example 1.5 processors == 1.5 physical core entitlements == ceil[1.5] = 2 vCPU == 16 logical OS CPUs (SMT=8)
+### Example 2 processors == 2 physical core entitlements == ceil[2] = 2 vCPU == 16 logical OS CPUs (SMT=8)
+bastion                     = {memory      = "16",   processors  = "1",    "count"   = 1}
+
+rhel_image_name             = "rhel-83-11242020"
+processor_type              = "shared"
+system_type                 = "s922"
+network_name                = "ocp-net"
+
```
Comment (yussufsh) [inline infra-node/var.tfvars:21]: vm_id and prefix can be added here.

File: infra-node/infra-node.tf (hunk 4 above)
Comment (sudeeshjohn) [reply infra-node/infra-node.tf:42]: should work , updating

**Review by yussufsh — APPROVED** [review]

**Review by bpradipt — APPROVED** [review]
/lgtm

**Comment by ltccci** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/ocp-power-automation/infra/pull/33#pullrequestreview-549025155" title="LGTM">bpradipt</a>*, *<a href="https://github.com/ocp-power-automation/infra/pull/33#issuecomment-742479028" title="Approved">mkumatag</a>*, *<a href="https://github.com/ocp-power-automation/infra/pull/33#" title="Author self-approved">sudeeshjohn</a>*, *<a href="https://github.com/ocp-power-automation/infra/pull/33#pullrequestreview-549013423" title="Approved">yussufsh</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=ocp-power-automation%2Finfra).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/ocp-power-automation/infra/blob/master/OWNERS)~~ [bpradipt,mkumatag,sudeeshjohn]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**